drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/handlers/HitPolicySelectorViewImpl.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.drools.workbench.screens.guided.dtable.client.handlers;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import com.google.gwt.dom.client.AnchorElement;
+import com.google.gwt.dom.client.Document;
+import com.google.gwt.dom.client.LIElement;
+import com.google.gwt.event.logical.shared.ValueChangeEvent;
+import com.google.gwt.event.logical.shared.ValueChangeHandler;
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.user.client.Event;
+import com.google.gwt.user.client.ui.Composite;
+import org.drools.workbench.models.guided.dtable.shared.model.GuidedDecisionTable52;
+import org.drools.workbench.screens.guided.dtable.client.resources.HitPolicyInternationalizer;
+import org.jboss.errai.common.client.dom.Node;
+import org.jboss.errai.common.client.dom.Span;
+import org.jboss.errai.common.client.dom.UnorderedList;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+
+@Dependent
+@Templated
+public class HitPolicySelectorViewImpl
+        extends Composite
+        implements HitPolicySelectorView {
+
+    @Inject
+    @DataField("selectedValue")
+    Span selectedValue;
+
+    @Inject
+    @DataField("listOfAvailableHitModes")
+    UnorderedList listOfAvailableHitModes;
+
+    @Override
+    public void setSelection( final GuidedDecisionTable52.HitPolicy hitPolicy ) {
+        selectedValue.setTextContent( HitPolicyInternationalizer.internationalize( hitPolicy ) );
+    }
+
+    @Override
+    public void addHitPolicyOption( final GuidedDecisionTable52.HitPolicy hitPolicy ) {
+        final AnchorElement anchor = Document.get()
```

Comment:
Hehe, I remember a time when making a dropdown was 1-3 lines and did not take 5 files and +100 lines of code (but not making the mistake of saying everything was better back then, one of those files is the test for example).

I have to say I am not convinced widget libraries are a bad thing. I guess the goal is to have the HTML handed over to us so we save time with UI. In the end we only get the huge changes as HTML and then end up spending more time than we save on the small daily changes like this. Also not thinking Errai UI is a bad thing, just not using Widgets adds a lot of styling and setups.

There is also more functions that can break here, when comparing to `new Dropdown()`. I can get the wrong styling ( I did ), leave out the event firing ( did that too ), but this was the first time I did this so that partially explains it.

Got this bit from the Uberfire examples, but maybe there is a better way and the examples need an update.